### PR TITLE
AP endpoints are now cached

### PR DESCRIPTION
**Pull request by annando**
Remote systems are polling these endpoints a lot. In the future we should increase the cache time and add some cache update when the content has changed.

### CODE REVIEW

File: src/Protocol/ActivityPub/Transmitter.php [hunk 1]
```diff
@@ -192,6 +201,10 @@ public static function getContacts(array $owner, array $rel, string $module, int
 		}
 
 		if (!$show_contacts) {
+			if (!empty($cachekey)) {
```
Comment (MrPetovan) [inline src/Protocol/ActivityPub/Transmitter.php:204]: According to the description of the parameter, I believe you should have checked the value of `$nocache` here.

File: src/Protocol/ActivityPub/Transmitter.php [hunk 2]
```diff
@@ -216,6 +229,10 @@ public static function getContacts(array $owner, array $rel, string $module, int
 			$data['orderedItems'] = $list;
 		}
 
+		if (!empty($cachekey)) {
```
Comment (MrPetovan) [inline src/Protocol/ActivityPub/Transmitter.php:232]: According to the description of the parameter, I believe you should have checked the value of `$nocache` here.

File: src/Protocol/ActivityPub/Transmitter.php [hunk 3]
```diff
@@ -293,27 +319,41 @@ public static function getOutbox(array $owner, int $page = null, string $request
...
```
Comment (MrPetovan) [inline src/Protocol/ActivityPub/Transmitter.php:322]: According to the description of the parameter, I believe you should have checked the value of `$nocache` here.

File: src/Protocol/ActivityPub/Transmitter.php [hunk 4]
```diff
@@ -362,6 +402,10 @@ public static function getFeatured(array $owner, int $page = null)
 
 		$data['orderedItems'] = $list;
 
+		if (!empty($cachekey)) {
```
Comment (MrPetovan) [inline src/Protocol/ActivityPub/Transmitter.php:405]: According to the description of the parameter, I believe you should have checked the value of `$nocache` here.

File: src/Protocol/ActivityPub/Transmitter.php (hunk 1 above)
Comment (annando) [reply src/Protocol/ActivityPub/Transmitter.php:204]: No. The parameter controls whether the cached value should be used or not. It is intentional that the cache is updated under any condition.

Comment (MrPetovan) [reply src/Protocol/ActivityPub/Transmitter.php:204]: Very well, but this is not what the description of the parameter suggests.

**Pull request merged**